### PR TITLE
[chore] Update jslib & state services to match

### PR DESCRIPTION
**Pull request by addisonbeck**
## Relevant work
* Duplicate PR: https://github.com/bitwarden/directory-connector/pull/211
* jslib update to storage schema: https://github.com/bitwarden/jslib/pull/611
  * This PR split the "state" object in storage up into several top level account keys, a globals key, and an authenticatedAccounts key.
* jslib update to ensure data is properly migrated when there isn't an actively logged in account: https://github.com/bitwarden/jslib/pull/615
  * This PR checked for a userId in storage, and if one isn't found saves all settings to a temporary key that are applied to the next authenticated account before being cleared.
* jslib update to fix key connector issues https://github.com/bitwarden/jslib/pull/616/files

## Type of change

- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
Essentially this is just a jslib update for the directory connector. 

Some changes needed to be made to account for recent key connector changes. 

Some other changes needed to be made to adopt the new storage structure passed down from the base jslib State service.

## Code changes
* Updated jslib
* Added logic to the StateMigrationService and StateService for managing temporary storage of settings and directory configurations if an active account isn't found.
* Restructured StateMigrationService and StateService logic to work with accounts as top level items without a parent "state" key.
* Removed LoginCommand params that were removed from jslib recently
* Added the AccountFactory param needed for the StateService to create accounts.

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/services/state.service.ts [hunk 1]
```diff
@@ -10,13 +10,15 @@ import { IConfiguration } from "src/models/IConfiguration";
 import { LdapConfiguration } from "src/models/ldapConfiguration";
 import { OktaConfiguration } from "src/models/oktaConfiguration";
 import { OneLoginConfiguration } from "src/models/oneLoginConfiguration";
+import { SyncConfiguration } from "src/models/syncConfiguration";
 
 import { LogService } from "jslib-common/abstractions/log.service";
+import { StateMigrationService } from "jslib-common/abstractions/stateMigration.service";
```
Comment (differsthecat) [inline src/services/state.service.ts:16]: Is there a reason the migration service is changing from the one in Directory Connector to the one in jslib here?

File: src/services/state.service.ts [hunk 2]
```diff
@@ -542,4 +561,12 @@ export class StateService extends BaseStateService<Account> implements StateServ
...
     this.accounts.next(this.state.accounts);
   }
+
+  protected async hasTemporaryStorage(): Promise<boolean> {
+    return (
+      (await this.storageService.has(keys.tempAccountSettings)) ||
+      (await this.storageService.has(keys.tempDirectorySettings)) ||
+      (await this.storageService.has(keys.tempDirectoryConfigs))
+    );
+  }
```
Comment (differsthecat) [inline src/services/state.service.ts:571]: Have you tested these changes locally to make sure the migrations are good? I can run through them if you'd like as I have a decent setup already if that's any help

File: src/services/stateMigration.service.ts [hunk 3]
```diff
@@ -80,73 +77,74 @@ export class StateMigrationService extends BaseStateMigrationService {
...
+    const directorySettings: DirectorySettings = {
+      directoryType: await this.get<DirectoryType>(Keys.directoryType),
+      organizationId: await this.get<string>(Keys.organizationId),
+      lastUserSync: await this.get<Date>(Keys.lastUserSync),
+      lastGroupSync: await this.get<Date>(Keys.lastGroupSync),
+      lastSyncHash: await this.get<string>(Keys.lastSyncHash),
+      syncingDir: await this.get<boolean>(Keys.syncingDir),
+      sync: await this.get<SyncConfiguration>(Keys.syncConfig),
+    };
+    //
```
Comment (differsthecat) [inline src/services/stateMigration.service.ts:124]: Extra comment line snuck in here

File: src/services/state.service.ts (hunk 2 above)
Comment (differsthecat) [reply src/services/state.service.ts:571]: Both Addison and I confirmed that everything is migrating over well!

**Review by differsthecat — DISMISSED** [review]

**Review by eliykat — APPROVED** [review]
The Key Connector related changes look good, thanks! Will let @differsthecat give the final review on everything else going on here.

**Comment by eliykat** [comment]
Oops, just noticed one other change that needs to be made - I'll push it directly in a sec.

File: src/services/stateMigration.service.ts (hunk 3 above)
Comment (addisonbeck) [reply src/services/stateMigration.service.ts:124]: This was intentional as a way to try and surround chunks of code with context. It didn't seem appropriate to break migrations out into multiple methods, except maybe inline ones like that clearKeys method. 

That said, you calling this out sort of proves it isn't a helpful way to group code - so let me know your thoughts on this and I'll address now or in a subsequent PR.

**Review by differsthecat — APPROVED** [review]
Looks good!

**Pull request merged**